packages/lodestar/src/chain/chain.ts

```diff
@@ -142,35 +142,29 @@ export class BeaconChain extends (EventEmitter as { new(): ChainEventEmitter })
 
     if(!await this.db.block.has(block.parentRoot)) {
       this.emit("unknownBlockRoot", block.parentRoot);
+      this.blockProcessingQueue.add(block);
+      return;
     }
 
-    if(block.slot <= this.latestState.slot) {
+    if(await this.db.block.has(blockHash)) {
+      this.logger.warn(`Block ${blockHash} existed already, no need to process it.`)
+      return;
+    }
+
+    const headBlockRoot = this.forkChoice.head();
+    const headBlock = await this.db.block.get(headBlockRoot);
+    if(block.slot <= headBlock.slot) {
```

Comment:
This should be roughly `if(block.slot <= this.forkChoice.finalized.slot)`

and the log message should be something like "block ${b} is not after the finalized checkpoint ${c}"